### PR TITLE
Module 4 - Implemented Clean Architecture on the project

### PR DESCRIPTION
**Pull request by LuciaMBandeo**
Implemented Clean Architecture on the project.

This is how the project structure looks:
image
image
image

### CODE REVIEW

**Review by LeonardoMolleker — APPROVED** [review]

File: lib/core/utils/constants/nums.dart
Comment (matiasilveiro-gl) [inline lib/core/utils/constants/nums.dart:1]: Super minor: I'd rename this class as `Dimens` or something more meaningful

**Review by matiasilveiro-gl — APPROVED** [review]
LGTM!: 🚀

**Review by Kevin-Guillen — COMMENTED** [review]
lgtm

File: lib/core/utils/constants/nums.dart
Comment (LuciaMBandeo) [reply lib/core/utils/constants/nums.dart:1]: Perfect! I agree with you, i'll change the name :)

**Review by Kevin-Guillen — APPROVED** [review]

File: lib/config/router/movie_router.dart [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class MovieRouter {
+  static const String movieList = 'MovieList';
+  static const String movieDetails = "MovieDetails";
```
Comment (FranciscoGaraventa) [inline lib/config/router/movie_router.dart]: Define the routes like` /movie_list` and `/movie_details`

Comment (FranciscoGaraventa) [reply lib/config/router/movie_router.dart]: And always you need to define an initial route that is `'/'`, for this particular case could be`'/movie_list'` but remember to change it when you add the splash screen.
Replace `'/movie_list'` to `'/'`

Comment (FranciscoGaraventa) [reply lib/config/router/movie_router.dart]: https://docs.flutter.dev/cookbook/navigation/named-routes

Comment (LuciaMBandeo) [reply lib/config/router/movie_router.dart]: Done!

**Review by FranciscoGaraventa — APPROVED** [review]
LG

**Pull request merged**